### PR TITLE
fetch reviews from db

### PR DESCRIPTION
**Pull request by gorakong**
added functionality: fetching & rendering of reviews from db.

@JayBo7 @VALE46GP @jcelder  Please review, thanks!

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -12,9 +12,13 @@ class App extends React.Component {
...
 	}
 
+	componentDidMount() {
+		this.getReviews();
+	}
+
 	getReviews() {
 		$.ajax({
-			url: '/',
+			url: '/reviews',
```
Comment (jcelder) [inline client/src/components/App.jsx]: @gorakong The spacing in this file, as well as other files in this commit, doesn't conform to Airbnb's style guidelines. Are you sure your linter is configured correctly?

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -36,7 +40,7 @@ class App extends React.Component {
 				<h1>hi from react</h1>
 				<Header />
 				<Ratings />
-				<Reviews reviewsData={this.getReviews()}/>
+				<Reviews reviewsData={this.state.reviews}/>
```
Comment (jcelder) [inline client/src/components/App.jsx]: @gorakong You should use object destructuring to pull the reviews property off of the state object.

File: client/src/components/Reviews.jsx [hunk 3]
```diff
@@ -1,13 +1,9 @@
-var Reviews = ({reviewsData}) => {
-	return (
-		<div>
-			{reviewsData}
-		</div>
-	)
-}
+var Reviews = ({reviewsData}) => (
+	<div>
+		{reviewsData.map((review, i) =>
```
Comment (jcelder) [inline client/src/components/Reviews.jsx]: @gorakong Do you need the index of a specific review for this map? If not, then there isn't really a reason to explicitly pass it into your callback.

File: server/model/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const con = require('../database/index.js');
+
+let fetchAll = function(cb) {
```
Comment (jcelder) [inline server/model/index.js]: Are you attempting to fetch all of the reviews or fetch all of the reviews for a specific listing? If you are attempting to fetch all of the reviews for a specific listing, then you will need to refactor this query.

File: server/app.js [hunk 5]
```diff
@@ -6,27 +6,18 @@ const bodyParser = require('body-parser');
 
 const path = require('path');
 
-const con = require('./database');
+const fetchAll = require('./model/index.js');
 
 app.use(express.static(path.join(__dirname, '/../client/dist')));
 app.use(bodyParser.json());
 
+app.get('/reviews', function(req, res) {
```
Comment (jcelder) [inline server/app.js]: @gorakong This endpoint does not match your listed endpoints in your [service plan](https://docs.google.com/document/d/19M5W_dSp8zcd9T9t4eLTKkGgW8KGT3Iwcu6RKzFmRxI).

**Review by jcelder — CHANGES_REQUESTED** [review]
@gorakong Let me know when you've made these changes.

File: client/src/components/App.jsx (hunk 1 above)
Comment (gorakong) [reply client/src/components/App.jsx]: @jcelder I have yet to change the syntax of my files to comply to my linter/Airbnb's style guidelines, but I will make those changes now. Thanks for the reminder!

File: client/src/components/App.jsx (hunk 2 above)
Comment (gorakong) [reply client/src/components/App.jsx]: @jcelder I have made the appropriate changes.

File: client/src/components/Reviews.jsx (hunk 3 above)
Comment (gorakong) [reply client/src/components/Reviews.jsx]: @jcelder It is indeed unnecessary! I put it in originally with the intent of using it as a key prop for each review object.

File: server/app.js (hunk 5 above)
Comment (gorakong) [reply server/app.js]: @jcelder Good catch--I have refactored my code to match the appropriate endpoint, & fixed the associated get request & responses.

File: server/model/index.js (hunk 4 above)
Comment (gorakong) [reply server/model/index.js]: @jcelder  query has been refactored to fetch reviews for a specific listing.

File: server/app.js [hunk 6]
```diff
@@ -6,27 +6,19 @@ const bodyParser = require('body-parser');
...
+const fetchAll = require('./model/index.js');
 
 app.use(express.static(path.join(__dirname, '/../client/dist')));
 app.use(bodyParser.json());
 
-
-app.get('/', (req, res) => {
-	console.log('req body is:', req.body);
-	res.send(get());
+app.get('/:listingId/reviews', (req, res) => {
```
Comment (jcelder) [inline server/app.js]: @gorakong Your endpoint still does not match the one listed in your service plan.

**Review by jcelder — CHANGES_REQUESTED** [review]

File: server/app.js (hunk 6 above)
Comment (gorakong) [reply server/app.js]: @jcelder  whoops! fixed!

Comment (jcelder) [reply server/app.js]: @gorakong Looks good!

**Review by jcelder — APPROVED** [review]

**Pull request merged**